demos/demo_stacking_simulation.py

```diff
@@ -0,0 +1,1143 @@
+			except:
+				J_inv = np.linalg.pinv(self.J.T)
+			F = J_inv * tau
+
+			direction = 'x'
+			idx = {'x':[0,3,6],
+				   'y':[1,4,7],
+				   'z':[2,5,8]}
+
+			if t%30==0:
+				degrees = []
+				for i in q:
+					degrees.append(math.degrees(i))
+
+				# print("Force applied along axis:",
+				# 		F[idx[direction][0]],
+				# 		F[idx[direction][1]],
+				# 		F[idx[direction][2]],
+				# 		"*****",
+				# 		F[idx[direction][0]]/force_applied*100, 
+				# 		F[idx[direction][1]]/force_applied*100, 
+				# 		F[idx[direction][2]]/force_applied*100)
+				# if self.is_grabbing:
+				# 	print("Q position is \n\t", degrees[:3], "\n\t", degrees[3:6], "\n\t", degrees[6:])
+				# 	q = self.calculate_q_compensation()
+				# 	degrees = []
+				# 	for i in q:
+				# 		degrees.append(math.degrees(i))
+				# 	print("Fixed Q position is \n\t", degrees[:3], "\n\t", degrees[3:6], "\n\t", degrees[6:])
+
+				# else:
+				# 	print("Q position is \n", degrees[:3], "\n", degrees[3:6], "\n", degrees[6:])
+				print("Q position is \n", degrees[:3], "\n", degrees[3:6], "\n", degrees[6:])
+
+				# print(force_applied)
+			
+				print("X_obsered:\n\t\t\t\t\t\t\t\t\t\t",
+						self.x_obs[:3].T, "\n\t\t\t\t\t\t\t\t\t\t",
+						self.x_obs[3:6].T, "\n\t\t\t\t\t\t\t\t\t\t",
+						self.x_obs[6:].T)
+
+				print("Torque obsered:\n",
+						tau[:3].T, "\n",
+						tau[3:6].T, "\n",
+						tau[6:].T)
+
+				# # # print("Percentage of force actually applied:",
+				# # # 		F[idx[direction][0]]/force_applied*100, 
+				# # # 		F[idx[direction][1]]/force_applied*100, 
+				# # # 		F[idx[direction][2]]/force_applied*100)
+
+				# print("Force applied along axis:",
+				# 			F[idx[direction][0]],
+				# 			F[idx[direction][1]],
+				# 			F[idx[direction][2]])
+
+
+				# # Plotting real-time values for the different terms of the total force
+				if self.is_grabbing:
+					position = self.Kp * (self.x_des - self.x_obs)
+					velocity = self.Kd * (self.v_des - self.v_obs)
+					global TIME
+
+					if self.plot_real_time:
+						plt.plot(TIME, F.data.tolist()[3][0], 'bo')
+						plt.plot(TIME, position.data.tolist()[3][0], 'y+')
+						plt.plot(TIME, velocity.data.tolist()[3][0], 'g+')
+						plt.plot(TIME, self.desired_external_force.data.tolist()[3][0], 'ro')
+						plt.pause(0.0001)
+
+					else:
+						self.plot_dict['Fo'].append(F.data.tolist()[3][0])
+						self.plot_dict['P'].append(position.data.tolist()[3][0])
+						self.plot_dict['V'].append(velocity.data.tolist()[3][0])
+						self.plot_dict['Fd'].append(self.desired_external_force.data.tolist()[3][0])
+
+					TIME += 1
+
+
+	def generate_plots(self):
+		print("Generating plots")
+		for time in range(len(self.plot_dict['Fo'])):
+			plt.plot(time, self.plot_dict['Fo'][time], 'bo')
+			plt.plot(time, self.plot_dict['P'][time], 'y+')
+			plt.plot(time, self.plot_dict['V'][time], 'g+')
+			plt.plot(time, self.plot_dict['Fd'][time], 'ro')
+			plt.pause(0.0001)
+			# plt.show()
+
+
+	
+
+
+	def get_desired_external_force(self):
```

Comment:
This function is called `get_...` but does not return anything.  Is this intended?
Note that Python those not give an error when you try to assign the return value, it will simply set it to None.  So this can easily lead to not so obvious bugs.

Reply:
Actually, you are trying to assign its return value below in line 464:

    desired_external_force = self.get_desired_external_force()